code/modules/reagents/Chemistry-Goon-420BlazeIt.dm

```diff
@@ -143,4 +143,356 @@ datum/reagent/crank/addiction_act_stage4(var/mob/living/M as mob)
 	required_reagents = list("diphenhydramine" = 1, "morphine" = 1, "cleaner" = 1, "potassium" = 1, "phosphorus" = 1, "fuel" = 1)
 	result_amount = 6
 	mix_message = "The mixture dries into a pale blue powder."
-	required_temp = 380
\ No newline at end of file
+	required_temp = 380
+
+/datum/reagent/methamphetamine
+	name = "Methamphetamine"
+	id = "methamphetamine"
+	description = "3x stun reduction per cycle, significant stamina regeneration buff, makes you really jittery, dramatically increases movement speed."
+	reagent_state = LIQUID
+	color = "#60A584" // rgb: 96, 165, 132
+	overdose_threshold = 20
+	addiction_threshold = 10
+	metabolization_rate = 0.6
+
+/datum/reagent/methamphetamine/on_mob_life(var/mob/living/M as mob)
+	if(!M) M = holder.my_atom
+	var/high_message = pick("You feel hyper.", "You feel like you need to go faster.", "You feel like you can run the world.")
+	if(prob(5))
+		M << "<span class='notice'>[high_message]</span>"
+	M.AdjustParalysis(-3)
+	M.AdjustStunned(-3)
+	M.AdjustWeakened(-3)
+	M.adjustStaminaLoss(-3)
+	M.status_flags |= GOTTAGOFAST
+	M.Jitter(3)
+	M.adjustBrainLoss(0.5)
+	M.emote(pick("twitch", "shiver"))
```

Comment:
should probably add a chance for either of these to happen, otherwise, it's going to make you twitch or shiver every single proc. I'd just use Hyperzine's old probabilities: if(prob(5)) M.emote(pick("twitch","blink_r","shiver"))